src/diffusers/pipelines/cogvideo/pipeline_cogvideox_image2video.py

```diff
@@ -726,6 +725,22 @@ def __call__(
         self._num_timesteps = len(timesteps)
 
         # 5. Prepare latents
+        # TODO: Only CogVideoX1.5-5B-I2V can use this method. Need to Change
+        def adjust_resolution_to_divisible(image_height, image_width, tgt_height, tgt_width, divisor=16):
+            # Step 1: Compare image dimensions with target dimensions
+            if image_height > tgt_height:
+                image_height = tgt_height
+            if image_width > tgt_width:
+                image_width = tgt_width
+
+            # Step 2: Ensure height and width are divisible by the divisor
+            image_height = (image_height // divisor) * divisor
+            image_width = (image_width // divisor) * divisor
+            return image_height, image_width
+
+        image_width, image_height = image.size[-2:]
+
+        height, width = adjust_resolution_to_divisible(image_height, image_width, height, width)
```

Comment:
```suggestion
```

I think this will not be required. We can mention it in the docs as well as add a comment saying that the image_height and image_width need to be divisible by 16. This check can also be added to the `check_inputs` method. The user needs to make sure to correctly pass the `height` and `width` parameters based on the input image dimension